nuxeo-admin-console-web/angular-app/src/app/features/sub-features/generic-multi-feature-layout/services/generic-multi-feature-utilities.service.ts

```diff
@@ -82,6 +84,32 @@ export class GenericMultiFeatureUtilitiesService {
   }
 
   insertParamInQuery(requestQuery: string, param: string) {
-    return requestQuery.replaceAll("{queryParam}", param);
+    return requestQuery.replaceAll("{query}", param);
+  }
+
+  buildRequestParams(
+    data: RequestParamType,
+    requestQuery: string,
+    inputForm: FormGroup
+  ): { requestUrl: string; requestParams: URLSearchParams } {
+    let requestUrl = "";
+    // Prepare request payload body
+    const requestParams = new URLSearchParams();
+    // Prepare body params object with dynamic parameters & their values entered as input
+    if (data["bodyParam"]) {
+      Object.keys(data["bodyParam"]).forEach((key) => {
+        if (key === GENERIC_LABELS.QUERY) {
+          requestParams.append(key, requestQuery);
+          return;
+        }
+        const paramValue = inputForm.get(key)?.value;
+        if (!paramValue) return;
+        requestParams.append(key, paramValue as string);
+      });
+    } else {
```

Comment:
add another if for queryParam rather than else. Handle negative scenario where both are not present.